### PR TITLE
Added itinerary drawing and ability to see user's live location on map

### PR DESCRIPTION
**Pull request by ethangraham2001**
#106, #100, #96, #95  
- Shortest paths are computed with Google Directions API
- User's current location can now be shown on the map
- Updated tests for parity with the new functionality in `MapViewModel`

## Problems / TODO
- Places directions don't work great for points that aren't snapped onto a road. This is likely due to us passing coordinates and not addresses to the directions API; will be updated in future sprints
- Maybe make the user location marker a bit prettier as now it is just a blue marker with standard shape
- Currently locations are hardcoded (for demoing purposes) but will be addressed in futures once itineraries can be dynamically created by users

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/model/location/SF_LOCATIONS.kt
Comment (imaneb942) [inline app/src/main/java/com/github/wanderwise_inc/app/model/location/SF_LOCATIONS.kt:1]: why was one point deleted?

File: app/src/main/java/com/github/wanderwise_inc/app/ui/map/PreviewItineraryScreen.kt
Comment (imaneb942) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/map/PreviewItineraryScreen.kt:1]: great modifications seems more cohesive right now

File: app/src/main/java/com/github/wanderwise_inc/app/model/location/SF_LOCATIONS.kt
Comment (ethangraham2001) [reply app/src/main/java/com/github/wanderwise_inc/app/model/location/SF_LOCATIONS.kt:1]: It wasn't snapped to the road, so google Directions APi wasn't computing the path properly. Removed for demoing purposes while bearing in mind that this is an issue that needs to be addressed down the line

**Review by imaneb942 — APPROVED** [review]
The modifications brought were interesting, the features also were cohesive and it was smart to start by gradually merging the branches before going to dev

**Pull request merged**